components/cli/pkg/commands/run.go

```diff
@@ -19,71 +19,58 @@
 package commands
 
 import (
-	"bufio"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
 	"os"
-	"os/exec"
-	"os/user"
 	"path"
 	"path/filepath"
 	"regexp"
-	"runtime"
 	"strings"
-	"sync"
-	"time"
 
-	"github.com/cellery-io/sdk/components/cli/pkg/constants"
+	"github.com/cellery-io/sdk/components/cli/ballerina"
+	"github.com/cellery-io/sdk/components/cli/cli"
 	"github.com/cellery-io/sdk/components/cli/pkg/image"
 	"github.com/cellery-io/sdk/components/cli/pkg/util"
-	"github.com/cellery-io/sdk/components/cli/pkg/version"
 )
 
 // RunRun starts Cell instance (along with dependency instances if specified by the user)
 // This also support linking instances to parts of the dependency tree
 // This command also strictly validates whether the requested Cell (and the dependencies are valid)
-
-const celleryEnvVar = "cellery_env_"
-
-func RunRun(cellImageTag string, instanceName string, startDependencies bool, shareDependencies bool,
-	dependencyLinks []string, envVars []string) {
-	spinner := util.StartNewSpinner("Extracting Cell Image " + util.Bold(cellImageTag))
-	parsedCellImage, err := image.ParseImageTag(cellImageTag)
-	if err != nil {
-		util.ExitWithErrorMessage("Error occurred while parsing cell image", err)
-	}
-	imageDir, err := ExtractImage(parsedCellImage, true, spinner)
-	if err != nil {
-		spinner.Stop(false)
-		util.ExitWithErrorMessage("Error occurred while extracting image", err)
-	}
-	defer func() {
-		err = os.RemoveAll(imageDir)
-		if err != nil {
-			spinner.Stop(false)
-			util.ExitWithErrorMessage("Error occurred while cleaning up", err)
-		}
+func RunRun(cli cli.Cli, cellImageTag string, instanceName string, startDependencies bool, shareDependencies bool,
+	dependencyLinks []string, envVars []string) error {
+	var err error
+	var parsedCellImage *image.CellImage
+	if parsedCellImage, err = image.ParseImageTag(cellImageTag); err != nil {
+		return fmt.Errorf("error occurred while parsing cell image, %v", err)
+	}
+	var imageDir string
+	if err = cli.ExecuteTask("Extracting cell image", "Failed to extract cell image",
+		"", func() error {
+			imageDir, err = ExtractImage(parsedCellImage, true)
+			return err
+		}); err != nil {
+		return err
+	}
+	defer func() error {
+		if err = os.RemoveAll(imageDir); err != nil {
+			return fmt.Errorf("error occurred while extracting image, %v", err)
```

Comment:
Is this error related to RemoveAll?

Reply:
Yes.

Reply:
But it does not say anything about the file removing

Reply:
Updated the error message to "error occurred while removing image".